jmespath-core/src/main/java/io/burt/jmespath/Adapter.java

```diff
@@ -32,6 +33,12 @@
   String toString(T value);
 
   /**
+   * Converts the argument to a Double, or null if the argument does not
+   * represent a number.
+   */
+  Double toDouble(T value);
```

Comment:
And when I say _needs_ to be able to return `null` I guess I mean that I'm not sure where type checking is going to happen yet. Perhaps it will be possible to guarantee that toX will only be called after isX.